src/XrdHttp.cmake

```diff
@@ -46,6 +47,7 @@ if( BUILD_HTTP )
     XrdServer
     XrdUtils
     XrdCrypto
+    XrdCl
```

Comment:
I'm fine with that dependency, but maybe @abh3 is not? Otherwise, we can put this typedef into a common library used by XrdCl and XrdHttp, for example XrdUtils?

Reply:
and the dependency is only for the ChunkInfo structure (and ChunkList, which is a vector of the former) as it had occurred to me reuse an exiting type of object, and on the client side this is used for almost exactly the purpose it is used here. I was going to use a utility function from XrdClUtils too, but it didn't do exactly what was needed. so there's really not much needed from XrdCl. maybe an alternative might be to use another structure from the server or other class, or just invent an own type somewhere in XrdHttp. (This PR already introduces a type XrdHttpReadRangeHandler::UserRange and UserRangeList for similar purpose).

Reply:
Yes, it's just a typedef, so you may add this typedef somewhere in XrdHttp --> I don't think it will cause any harm ;)

Thanks for all your work!

Reply:
Likely, the best long term solution is to move XrdClChunklist to the utility package and use it both in the client and the http server part. What do you think? I know it's more work, sigh.

Reply:
I looked around and saw he have XrdOucIOVec, which is used in several places server/plugin side. It's quite similar to XrdCl::ChunkInfo. So I updated this PR to use that (in fact XrdOucIOVec2, as that has a useful constructor added). The missing piece was a typedef for a container of those (to correspond to ChunkList); so I added a  typedef for a std::vector<XrdOucIOVec2> in XrdHttpUtils.

We could move ChunkInfo/ChunkList to the utility package, and possibly try moving some uses of XrdOucIOVec to that, or simply kept both. But I thought perhaps it might be better to leave that out of this PR, given it seems quite alright to use XrdOucIOVec now.

Reply:
Thanks David! I am perfectly fine with what you did :)